### PR TITLE
Support GHC 9.4

### PR DESCRIPTION
**Pull request by parsonsmatt**
Before submitting your PR, check that you've:

- [x] Bumped the version number
- [ ] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [ ] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddocks for new, public APIs

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by ysangkok** [comment]
@parsonsmatt This passes the test suite even without any `allow-newer`, `constraints` or `source-repository-package`. So everything below `packages` in `cabal.project` can be removed.

**Comment by parsonsmatt** [comment]
MacOS is failing due to "Can't find `Cocoa`" ??

**Comment by ysangkok** [comment]
Possibly the CI runners switched to MacOS Big Sur, making them fail due to [GHC bug 18446](https://gitlab.haskell.org/ghc/ghc/-/issues/18446). The logs for lts-16 (GHC 8.8) doesn't show for me, but from the linked issue it looks like the oldest backport was for 8.10.

**Comment by parsonsmatt** [comment]
Yeah, that appears to be what's up. I disabled the MacOS runners for those GHCs, which is Unfortunate, but 🤷🏻

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -26,6 +26,14 @@ jobs:
         exclude:
           - os: windows-latest
             args: "--resolver nightly"
+          - os: macos-latest
```
Comment (snoyberg) [inline .github/workflows/tests.yml:29]: Unfortunate but necessary 👍

File: yesod-bin/ChangeLog.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # ChangeLog for yesod-bin
 
+## 1.6.2.2
+
+* Support Cabal 3.7 [#1769](https://github.com/yesodweb/yesod/pull/1769)
```
Comment (snoyberg) [inline yesod-bin/ChangeLog.md:5]: Is this supposed to be 3.8?

**Review by snoyberg — APPROVED** [review]

File: yesod-bin/ChangeLog.md (hunk 2 above)
Comment (parsonsmatt) [reply yesod-bin/ChangeLog.md]: Good catch, thanks!

**Comment by parsonsmatt** [comment]
@snoyberg or @MaxGabriel could you put together a release for `yesod` and `yesod-bin`?

**Comment by snoyberg** [comment]
Release :) I had to futz with GPG first, apparently my signing key expired today, and error messages aren't exactly helpful 😡

**Pull request merged**